wallet/src/de/schildbach/wallet/ui/scan/ScanActivity.java

```diff
@@ -99,23 +101,34 @@ public final class ScanActivity extends AbstractWalletActivity
 
     public static void startForResult(final Activity activity, @Nullable final View clickView, final int requestCode) {
         if (clickView != null) {
-            final int[] clickViewLocation = new int[2];
-            clickView.getLocationOnScreen(clickViewLocation);
-            final Intent intent = new Intent(activity, ScanActivity.class);
-            intent.putExtra(ScanActivity.INTENT_EXTRA_SCENE_TRANSITION_X,
-                    (int) (clickViewLocation[0] + clickView.getWidth() / 2));
-            intent.putExtra(ScanActivity.INTENT_EXTRA_SCENE_TRANSITION_Y,
-                    (int) (clickViewLocation[1] + clickView.getHeight() / 2));
-            final ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(activity, clickView,
-                    "transition");
```

Comment:
Refactored to make use of the `registerForActivityResult` instead of the obsolete `onActivityResult`